### PR TITLE
 Added missing documentation of header

### PR DESCRIPTION
**Pull request by polluks**
Maybe device.h should include stddef.h

### CODE REVIEW

File: doc/funcref.sgml [hunk 1]
```diff
@@ -3450,6 +3458,72 @@ returns one of the constants<itemize>
...
+
+
+<sect1>getdevicedir<label id="getdevicedir"><p>
+
+<quote>
+<descrip>
+<tag/Function/Get device directory.
+<tag/Header/<tt/<ref id="device.h" name="device.h">/
+<tag/Declaration/<tt/char* __fastcall__ getdevicedir (unsigned char device, char* buf, size_t size);/
+<tag/Description/The function returns the current directory of <tt/device/.
```
Comment (oliverschmidt) [inline doc/funcref.sgml]: This isn't correct. It's rather "The function returns the directory representing 'device'. It allows to access the device on filesystem level by calling chdir() with the directory returned."

File: doc/funcref.sgml [hunk 2]
```diff
@@ -3450,6 +3458,72 @@ returns one of the constants<itemize>
...
+be used in presence of a prototype.
+</itemize>
+<tag/Availability/cc65
+<tag/See also/
+<ref id="getcwd" name="getcwd">,
+<ref id="getcurrentdevice" name="getcurrentdevice">,
+<ref id="getfirstdevice" name="getfirstdevice">,
+<ref id="getnextdevice" name="getnextdevice">
+<tag/Example/<verb>
+puts (getdevicedir (getcurrentdevice (), buf, sizeof buf));
```
Comment (oliverschmidt) [inline doc/funcref.sgml]: This isn't a great example as the result is (most of the time) identical to getcwd().
Better `chdir (getdevicedir (device, buf, sizeof (buf)));`.

File: doc/funcref.sgml [hunk 3]
```diff
@@ -3450,6 +3458,72 @@ returns one of the constants<itemize>
...
+<tag/Notes/<itemize>
+<item>
+</itemize>
+<tag/Availability/cc65
+<tag/See also/
+<ref id="getdevicedir" name="getdevicedir">,
+<ref id="getfirstdevice" name="getfirstdevice">,
+<ref id="getnextdevice" name="getnextdevice">
+<tag/Example/<verb>
+puts (getdevicedir (getcurrentdevice (), buf, sizeof buf));
```
Comment (oliverschmidt) [inline doc/funcref.sgml]: This isn't a great example as the result is (most of the time) identical to getcwd().
Better `dio_open (getcurrentdevice ());`.

File: doc/funcref.sgml [hunk 4]
```diff
@@ -3450,6 +3458,72 @@ returns one of the constants<itemize>
...
 
 
+<sect1>getcurrentdevice<label id="getcurrentdevice"><p>
+
+<quote>
+<descrip>
+<tag/Function/Get current device.
+<tag/Header/<tt/<ref id="device.h" name="device.h">/
+<tag/Declaration/<tt/unsigned char getcurrentdevice (void);/
+<tag/Description/The function returns the current device.
```
Comment (oliverschmidt) [inline doc/funcref.sgml]: Please add "It allows to access the current device with the 'Low-level disk I/O API' or cbm_* I/O functions requiring a 'device' parameter."

**Comment by polluks** [comment]
Include std header:
What about size_t of stddef.h and device.h?
time.h seems also to be a candidate...

**Comment by greg-king5** [comment]
I think that "device,h" should define `size_t` independently (in the same way that "6502.h", "stddef.h", "stdlib.h", "time.h", and "unistd.h" do it).

**Comment by polluks** [comment]
Indeed, including a whole file for a single typedef is pretty much overhead.
But time.h defines NULL and size_t itself, this is exactly what stddef.h is made for!

**Pull request merged**